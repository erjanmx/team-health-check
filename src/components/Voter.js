import React, { Component } from "react";

class Voter extends Component {
  componentDidMount() {
    /*global Ably*/
    const channel = this.channel = Ably.channels.get(this.props.channelUuid);

    channel.attach();    
    channel.once("attached", () => {
      channel.presence.enter();
    });
  }

  handleVote(vote) {
    this.channel.publish('add_vote', { vote });
  }

  render() {
    return (
      <div className="has-text-centered">
        <h4 className="title is-4">{this.props.channelUuid}</h4>
        <div className="columns" style={{ fontSize: '7em' }}>
          <div className="column">
            <i onClick={(e) => this.handleVote(1, e)} class='far fa-smile'></i>
          </div>
          <div className="column">
            <i onClick={(e) => this.handleVote(0, e)} class='far fa-meh'></i>
          </div>
          <div className="column" >
            <i onClick={(e) => this.handleVote(-1, e)} class='far fa-frown'></i>
          </div>
        </div>
      </div>
    );
  }
}

export default Voter;
