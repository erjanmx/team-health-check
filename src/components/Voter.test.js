import React from 'react';
import { shallow, mount } from "enzyme";
import Voter from './Voter';

it("matches the snapshot", () => {
  const wrapper = shallow(<Voter channelUuid={1000} />);

  expect(wrapper).toMatchSnapshot();
});

it("changes icon class on click", () => {
  const wrapper = mount(<Voter channelUuid={1000} />);

  const firstIconDiv = wrapper.find('div .column').at(0);

  firstIconDiv.simulate('click');

  expect(firstIconDiv.childAt(0).html()).toBe('<i style="color: green;" class="fas fa-smile"></i>');
});

it("calls handler when icon clicked", () => {
  const wrapper = mount(<Voter channelUuid={1000} />);

  const spy = jest.spyOn(wrapper.instance(), 'handleVote');

  wrapper.find('div .column').at(1).simulate('click');

  expect(spy).toHaveBeenCalledWith(0, expect.anything());
});

it("handles reset updates", () => {
  const wrapper = shallow(<Voter channelUuid={1000} />);
  const instance = wrapper.instance();

  wrapper.setState({ vote: 1 });

  instance.handleSubscribe({ name: 'reset' });

  expect(wrapper.state().vote).toBe(null);
});
