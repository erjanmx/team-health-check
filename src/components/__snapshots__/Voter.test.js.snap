// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Voter
    channelUuid={1000}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <ChannelHeader
            channelUuid={1000}
          />
        </div>,
        <div
          className="columns"
          style={
            Object {
              "fontSize": "7em",
            }
          }
        >
          <div
            className="column"
            onClick={[Function]}
          >
            <Icon
              isSelected={false}
              type={1}
            />
          </div>
          <div
            className="column"
            onClick={[Function]}
          >
            <Icon
              isSelected={false}
              type={0}
            />
          </div>
          <div
            className="column"
            onClick={[Function]}
          >
            <Icon
              isSelected={false}
              type={-1}
            />
          </div>
        </div>,
      ],
      "className": "has-text-centered",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ChannelHeader
            channelUuid={1000}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "channelUuid": 1000,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="column"
              onClick={[Function]}
            >
              <Icon
                isSelected={false}
                type={1}
              />
            </div>,
            <div
              className="column"
              onClick={[Function]}
            >
              <Icon
                isSelected={false}
                type={0}
              />
            </div>,
            <div
              className="column"
              onClick={[Function]}
            >
              <Icon
                isSelected={false}
                type={-1}
              />
            </div>,
          ],
          "className": "columns",
          "style": Object {
            "fontSize": "7em",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Icon
                isSelected={false}
                type={1}
              />,
              "className": "column",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "isSelected": false,
                "type": 1,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Icon
                isSelected={false}
                type={0}
              />,
              "className": "column",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "isSelected": false,
                "type": 0,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Icon
                isSelected={false}
                type={-1}
              />,
              "className": "column",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "isSelected": false,
                "type": -1,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <ChannelHeader
              channelUuid={1000}
            />
          </div>,
          <div
            className="columns"
            style={
              Object {
                "fontSize": "7em",
              }
            }
          >
            <div
              className="column"
              onClick={[Function]}
            >
              <Icon
                isSelected={false}
                type={1}
              />
            </div>
            <div
              className="column"
              onClick={[Function]}
            >
              <Icon
                isSelected={false}
                type={0}
              />
            </div>
            <div
              className="column"
              onClick={[Function]}
            >
              <Icon
                isSelected={false}
                type={-1}
              />
            </div>
          </div>,
        ],
        "className": "has-text-centered",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ChannelHeader
              channelUuid={1000}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "channelUuid": 1000,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="column"
                onClick={[Function]}
              >
                <Icon
                  isSelected={false}
                  type={1}
                />
              </div>,
              <div
                className="column"
                onClick={[Function]}
              >
                <Icon
                  isSelected={false}
                  type={0}
                />
              </div>,
              <div
                className="column"
                onClick={[Function]}
              >
                <Icon
                  isSelected={false}
                  type={-1}
                />
              </div>,
            ],
            "className": "columns",
            "style": Object {
              "fontSize": "7em",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Icon
                  isSelected={false}
                  type={1}
                />,
                "className": "column",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "isSelected": false,
                  "type": 1,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Icon
                  isSelected={false}
                  type={0}
                />,
                "className": "column",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "isSelected": false,
                  "type": 0,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Icon
                  isSelected={false}
                  type={-1}
                />,
                "className": "column",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "isSelected": false,
                  "type": -1,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
